internal/provider/restore_resource.go

```diff
@@ -0,0 +1,353 @@
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var plan Restore
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var objects []client.RestoreItem
+	for _, object := range plan.Objects {
+		objects = append(objects, client.RestoreItem{
+			Database: object.Database,
+			Schema:   object.Schema,
+			Table:    object.Table,
+		})
+	}
+
+	var restoreOpts *RestoreOpts
+	if plan.RestoreOpts != nil {
+		restoreOpts = &client.RestoreOpts{
+			NewDbName:              plan.RestoreOpts.NewDBName,
+			IntoDb:                 plan.RestoreOpts.IntoDB,
+			SkipLocalitiesCheck:    plan.RestoreOpts.SkipLocalitiesCheck,
+			SkipMissingForeignKeys: plan.RestoreOpts.SkipMissingForeignKeys,
+			SkipMissingSequences:   plan.RestoreOpts.SkipMissingSequences,
+			SchemaOnly:             plan.RestoreOpts.SchemaOnly,
+		}
+	}
+
+	createRestoreReq := &client.CreateRestoreRequest{
+		DestinationClusterId: plan.DestinationClusterID.ValueString(),
+		BackupId:             plan.BackupID.ValueString(),
+		SourceClusterId:      plan.SourceClusterID.ValueString(),
+		Type:                 client.RestoreType(plan.Type.ValueString()),
+		Objects:              objects,
+		RestoreOpts:          restoreOpts,
+	}
+
+	traceAPICall("CreateRestore")
+	restoreObj, _, err := r.provider.service.CreateRestore(ctx, createRestoreReq)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating restore",
+			fmt.Sprintf("Could not create restore: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	// Poll until restore operation completes.
+	err = retry.RetryContext(ctx, createRestoreEndpointTimeout,
+		waitForRestoreReadyFunc(ctx, plan.DestinationClusterID.ValueString(), restoreObj.Id.ValueString(), r.provider.service, restoreObj))
+	if err != nil {
+		resp.Diagnostics.AddError("restore operation timed out and may have failed", formatAPIErrorMessage(err))
+		return
+	}
+
+	var state Restore
+	loadRestoreToTerraformState(restoreObj, &state)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *restoreResource) Read(
+	ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse,
+) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var state Restore
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if state.ID.IsNull() {
+		return
+	}
+
+	restoreID := state.ID.ValueString()
+
+	// In case this was an import, validate the ID format.
+	if !uuidRegex.MatchString(restoreID) {
+		resp.Diagnostics.AddError(
+			"Unexpected restore ID format",
+			fmt.Sprintf("'%s' is not a valid restore ID format. Expected UUID.", restoreID),
+		)
+		return
+	}
+
+	traceAPICall("GetRestore")
+	restoreObj, httpResp, err := r.provider.service.GetFolder(ctx, restoreID)
```

Comment:
I assume this is  a copy paste issue.
```suggestion
	traceAPICall("GetRestore")
	restoreObj, httpResp, err := r.provider.service.GetRestore(ctx, restoreID)
```